mkosi/config.py

```diff
@@ -844,8 +848,26 @@ class MkosiConfig:
     def output_dir_or_cwd(self) -> Path:
         return self.output_dir or Path.cwd()
 
-    def workspace_dir_or_cwd(self) -> Path:
-        return self.workspace_dir or Path.cwd()
+    def workspace_dir_or_default(self) -> Path:
+        if self.workspace_dir:
+            return self.workspace_dir
+
+        if (cache := os.getenv("XDG_CACHE_HOME")) and Path(cache).exists():
+            return Path(cache)
+
+        # If there's a cache directory and we're running from /home, we want to use a workspace directory in /home as
+        # well as /home might be on a separate partition or subvolume which means that to take advantage of reflinks
+        # and such, the workspace directory has to be on the same partition/subvolume.
+        if (
+            (home := os.getenv("HOME")) and
+            "/home" in home and
+            self.cache_dir and
+            self.cache_dir.is_relative_to(home) and
+            (Path(home) / ".cache").exists()
+        ):
+            return Path(home) / ".cache"
+
+        return Path("/var/tmp")
```

Comment:
I'm curious why this last fallback went (back?) to `/var/tmp` and not `/tmp`? The latter is often `tmpfs`, the former never.

Afraid to use too much RAM maybe?

More context in:
- https://github.com/pmem/run_qemu/pull/116

Reply:
@marc-hb Indeed, we write the entire image to the workspace directory, which can potentially be multiple gigabytes, we don't want to put all of that in RAM.